src/store/job/status/actions.js

```diff
@@ -0,0 +1,159 @@
+import { createAction } from 'redux-actions';
+import { lockApp, unlockApp } from '../../application/actions';
+import * as JobService from '../../../services/jobService';
+import { setJob } from '../actions';
+import { addPdfFile } from '../../pdfFiles/actions';
+import { uploadFile as uploadFilePOST } from '../../../services/fileService';
+import { JOB_ID, JOB_STATUS } from '../../constants';
+
+const startStep = createAction('STEP_START');
+const finishStep = createAction('STEP_FINISH');
+const setPercentage = createAction('PERCENTAGE_SET');
+const resetStatus = createAction('JOB_RESET');
+const setStatus = createAction('STATUS_SET');
+
+export const validateJob = id => async (dispatch, getState) => {
+    const profile = getState().jobSettings.profile;
+    let fileData = getState().pdfFiles[0];
+    let job;
+    dispatch(resetStatus());
+    dispatch(lockApp(!id));
+    try {
+        // Create job
+        job = await create(dispatch, profile, id);
+        dispatch(unlockApp());
+        dispatch(setPercentage(25));
+        if (!job.tasks?.length) {
+            // Upload file
+            const uploadedFileProps = await upload(dispatch, fileData);
+            dispatch(setPercentage(50));
+            // Update job
+            job = await update(dispatch, job, uploadedFileProps);
+        } else {
+            skipStep(dispatch, 'FILE_UPLOAD');
+            skipStep(dispatch, 'JOB_UPDATE');
+        }
+        dispatch(setPercentage(60));
+        // execute job
+        if (job.status !== JOB_STATUS.FINISHED) {
+            await execute(dispatch, job);
+        } else {
+            skipStep(dispatch, 'JOB_EXECUTE');
+        }
+        dispatch(setPercentage(100));
+        dispatch(setStatus(JOB_STATUS.FINISHED));
+    } catch (e) {
+        dispatch(setStatus(JOB_STATUS.ERROR));
+    } finally {
+        dispatch(unlockApp());
+    }
+};
+
+const create = async (dispatch, profile, id) => {
+    let job;
+    dispatch(startStep('JOB_CREATE'));
+    if (id) {
+        job = await getJob(id);
```

Comment:
Better split job create and get. Also I would split the logic for defining whether we need to create new job from scratch or we need to continue existing one.

E.g. we can parameterize `validate` action with optional parameter `completedSteps`. `getJob` should be called **before** calling `validate` and based on the structure of retrieved job a decision on which we should continue from should be made and `completedSteps` parameter constructed. After that in validate when we about to start some step we can check if it's complete and if so - just skip it. We can even put this logic into "wrapper function":
```
const createStep = (key, percent, stepFn) => async (dispatch, getState, completedSteps) => {
    dispatch(startStep(key));

    if (!completedSteps.includes(key)) {
        await stepFn(dispatch, getState);
    }    

    dispatch(finishStep(key, percentage)); // this should update both percentage and last step
}

const createJob = createStep('JOB_CREATE', 25, (dispatch, getState) => {
    // creation logic goes here. 

    // Note that it should be void and put necessary info into state if that needs sharing. 
    // This way it doesn't matter where the info is coming from - from previous step execution 
    //    or it was restored by `getJob` call made before validate
});

const validate = (completedSteps = []) => (dispatch, getState) => {
    try {
        createJob(dispatch, getState, completedSteps);
        uploadFile(dispatch, getState, completedSteps);
        updateJob(dispatch, getState, completedSteps);
        startJob(dispatch, getState, completedSteps);
    } finally {
        //cleanup
    }
};
```

PS: this starts taking too long, so we can postpone proper restoration logic for the future and handle only cases without interruption for now. So in our simplified implementation when we refresh the page we assume that validation was already started and we don't need to call `validate` at all. It's OK that progress info won't be restored for now.